src/core/thread/network_data_leader.cpp

```diff
@@ -110,8 +110,14 @@ otError LeaderBase::GetBackboneRouterPrimary(BackboneRouter::BackboneRouterConfi
     for (const NetworkDataTlv *start                                                      = serviceTlv->GetSubTlvs();
          (serverTlv = FindTlv<ServerTlv>(start, serviceTlv->GetNext())) != nullptr; start = serverTlv->GetNext())
     {
-        const BackboneRouterServerData *serverData =
-            reinterpret_cast<const BackboneRouterServerData *>(serverTlv->GetServerData());
+        const BackboneRouterServerData *serverData;
+
+        if (serverTlv->GetServerDataLength() < sizeof(BackboneRouterServerData))
```

Comment:
Can the length be equal?
```suggestion
        if (serverTlv->GetServerDataLength() <= sizeof(BackboneRouterServerData))
```

Reply:
I think if the server datalen is equal then if it fine to cast and parse it and we don't want to skip it.